src/Cache.cpp

```diff
@@ -2962,13 +2983,21 @@ Cache::roomNamesAndAliases()
                 alias = aliases->content.alias;
             }
 
+            auto parentId   = getParentRoomIdsWithTxn(room_id_str);
+            auto parentName = std::string{};
+
+            if (parentId) {
+                parentName = getRoomName(*parentId);
+            }
+
             result.push_back(RoomNameAlias{
               .id              = std::move(room_id_str),
               .name            = std::move(info.name),
               .alias           = std::move(alias),
               .recent_activity = info.approximate_last_modification_ts,
               .is_tombstoned   = info.is_tombstoned,
               .is_space        = info.is_space,
+              .parent          = std::move(parentName),
```

Comment:
Can we load this information lazily when it is accessed first in the RoomsModel instead? Looking up this data seems to add quite a lot of overhead to constructing the fuzzy search structure, but we only need it when displaying specific results,so we can really look it up only when showing a certain entry.

Reply:
I have a few deadlines coming up. I'll make the fixups in a few weeks :)